File: core/src/io/cucumber/model/base/Button.java
package io.cucumber.model.base;

import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.math.Rectangle;
import com.badlogic.gdx.math.Vector2;

import io.cucumber.model.bound.RectangleBound;

public class Button extends Actor<Rectangle> {

    protected Texture currentTexture;

    public Button(float x, float y, float width, float height, Texture currentTexture) {
        super(new RectangleBound(x, y, width, height));
        this.currentTexture = currentTexture;
    }

    @Override
    public void draw(Batch batch, float parentAlpha) {
        batch.draw(
                currentTexture,
                getX(),
                getY(),
                getWidth(),
                getHeight()
        );
    }

    public boolean isTouched(Vector2 position) {
        return contains(position);
    }
}

File: core/src/io/cucumber/model/base/Button.kt
package io.cucumber.model.base

import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.math.Rectangle

class Button(
    x: Float,
    y: Float,
    width: Float,
    height: Float,
    texturePath: String
) {

    val texture: Texture = Texture(texturePath)
    val bound: Rectangle = Rectangle(x, y, width, height)


    fun isTouched(x: Float, y: Float): Boolean {
        return bound.contains(x, y)
    }

}